packages/svelte/src/internal/client/dom/elements/transitions.js

```diff
@@ -311,6 +311,9 @@ function animate(element, options, counterpart, t2, callback) {
 				if (animation.startTime !== null && animation.currentTime !== null) {
 					throw e;
 				}
+				if (!element.isConnected) {
+					callback?.();
+				}
```

Comment:
this change could use a comment, it's not clear why we'd call the callback for an errored animation but only if the element isn't connected